inventory_management_system_api/repositories/item.py

```diff
@@ -103,3 +103,17 @@ def list(self, system_id: Optional[str], catalogue_item_id: Optional[str]) -> Li
 
         items = self._items_collection.find(query)
         return [ItemOut(**item) for item in items]
+
+    def update(self, item_id: str, item: ItemIn) -> ItemOut:
+        """
+        Update an item by its ID in a MongoDB database.
+
+        :param item_id: The ID of the item to update.
+        :param item: The item containing the update data.
+        :return: The updated item.
+        """
+        item_id = CustomObjectId(item_id)
+        logger.info("Updating item with ID: %s in the database", item_id)
+        self._items_collection.update_one({"_id": item_id}, {"$set": item.model_dump()})
+        item = self.get(str(item_id))
```

Comment:
We also need checks when changing the parent catalogue item/system to at the very least ensure they exist (see catalogue categories/systems). #153 should probably also do the same for catalogue items as it is currently missing.